packages/api/tests/lambdas/test-write-db-dlq-records-to-s3.js

```diff
@@ -107,16 +98,765 @@ test.serial('write-db-dlq-records-to-s3 throws error if system bucket is not def
+        body: JSON.stringify({
+          time: 'aTime',
+          detail: {
+            status: 'SUCCEEDED',
+            output: JSON.stringify({
+              meta: {
+                collection: { name: 'aName', version: '12' },
+                provider: {
+                  id: 'abcd',
+                  protocol: 'cheesy',
+                  host: 'excellent',
+                },
+              },
+              payload: { ss: [{ granuleId: 'a' }, { granuleId: 'b' }] },
+            }),
+            executionArn: 'execArn',
+            stateMachineArn: 'SMArn',
+          },
+        }),
+        error: 'anError',
+        collectionId: 'aName___12',
+        providerId: 'abcd',
+        executionArn: 'execArn',
+        stateMachineArn: 'SMArn',
+        granules: null,
+        time: 'aTime',
+        status: 'SUCCEEDED',
+      },
+    }, {
+      mangled: {
+        messageId: 'a',
+        eventSource: 'aws:sqs',
+        body: JSON.stringify({
+          detail: {
+            status: 'SUCCEEDED',
+            output: JSON.stringify({
+              meta: {
+                collection: { name: 'aName', version: '12' },
+                provider: {
+                  id: 'abcd',
+                  protocol: 'cheesy',
+                  host: 'excellent',
+                },
+              },
+              payload: { granules: [{ granuleId: 'a' }, { granuleId: 'b' }] },
+            }),
+            executionArn: 'execArn',
+            stateMachineArn: 'SMArn',
+          },
+        }),
+        error: 'anError',
+      },
+      expected: {
+        messageId: 'a',
+        eventSource: 'aws:sqs',
+        body: JSON.stringify({
+          detail: {
+            status: 'SUCCEEDED',
+            output: JSON.stringify({
+              meta: {
+                collection: { name: 'aName', version: '12' },
+                provider: {
+                  id: 'abcd',
+                  protocol: 'cheesy',
+                  host: 'excellent',
+                },
+              },
+              payload: { granules: [{ granuleId: 'a' }, { granuleId: 'b' }] },
+            }),
+            executionArn: 'execArn',
+            stateMachineArn: 'SMArn',
+          },
+        }),
+        error: 'anError',
+        collectionId: 'aName___12',
+        providerId: 'abcd',
+        executionArn: 'execArn',
+        stateMachineArn: 'SMArn',
+        granules: ['a', 'b'],
+        time: null,
+        status: 'SUCCEEDED',
+      },
     },
-  }),
-  executionName);
+  ];
+  const results = await Promise.all(
+    messages.map((message) => hoistCumulusMessageDetails(message.mangled))
+  );
+  results.forEach((result, index) => {
+    t.deepEqual(result, messages[index].expected);
+  });
 });
 
-test('determineExecutionName returns "unknown" if getExecutionName throws error', (t) => {
-  t.is(determineExecutionName({}), 'unknown');
+test('hoistCumulusMessageDetails handles varying degrees of sqsMessage nestedness', async (t) => {
```

Comment:
Maybe I'm missing or misunderstanding something, but in the test validation from here down, I'm not seeing an `error` in the messages being compared against.

Reply:
@Charles I'm pretty sure error is a field in hoistCumulusMessageDetails, and in the fixture as an `error` key , for example on `763`/`724`?

Reply:
error is present in those messages, but having up above verified that it *is* passed through, it should simply pass through